Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,311 @@
+    }
+
+    /**
+     * Responsible to disconnect with the broker.
+     *
+     * @throws \Exception
+     * @return void
+     */
+    public function disconnect()
+    {
+        if ($this->validateConnection()) {
+            $this->stream->close();
+        }
+    }
+
+    /**
+     * Creates the QueueMessage object.
+     *
+     * @return QueueMessageInterface
+     */
+    public function createQueueMessage(): QueueMessageInterface
+    {
+        $msg = new QueueMessage();
+        $msg->setContext(new Context());
+
+        return $msg;
+    }
+
+    /**
+     * Verifies if there is some connection opened with the broker.
+     *
+     * @return bool
+     */
+    public function isConnected(): bool
+    {
+        return $this->validateConnection() && $this->stream->isConnected();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function destroy(string $consumerTag)
+    {
+        $this->channel->basic_cancel($consumerTag);
+        $this->channel = null;
+        $this->disconnect();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function consume(string $consumerTag, string $queueName, callable $callback = null)
+    {
+        $this->declareChannel();
+        $this->declareQueue($queueName);
+        $this->channel->basic_consume($queueName, $consumerTag, false, false, false, false, $callback);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function ack(QueueMessageInterface $message = null)
+    {
+        $this->channel->basic_ack($message->getMessageId());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function nack(QueueMessageInterface $message = null)
+    {
+        $this->channel->basic_nack($message->getMessageId());
+    }
+
+    /**
+     * Creates a message and prepare the content to publish.
+     *
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    public function send(QueueMessageInterface $message, $destination = null, array $arguments = []): bool
+    {
+        $this->declareChannel();
+        $this->declareQueue($message->getQueue(), AMQP_DURABLE, $arguments);
+        $messageBody = $this->getSerializer()->serialize($message, $this->format);
+        $messageHeaders = new AMQPTable($message->getHeaders());
+        $amqpMessage = new AMQPMessage($messageBody);
+        $amqpMessage->set('application_headers', $messageHeaders);
+        $this->channel->basic_publish($amqpMessage, self::EXCHANGE_NAME, $message->getQueue());
+
+        return true;
+    }
+
+    /**
+     * Declares the queue to drive the message.
+     *
+     * @param string $queueName The name of the queue
+     * @param int $durable
+     * @param array $arguments See AMQPQueue::setArguments()
+     *
+     * @return array|null
+     */
+    public function declareQueue(string $queueName, int $durable = AMQP_DURABLE, array $arguments = [])
+    {
+        return $this->channel->queue_declare($queueName, false, $durable, false, false, false, new AMQPTable([$arguments]));
```

Comment:
Is it expected in `AMQPTable`to have an array of an array?

Reply:
No, just an array.